js/plates-model/fields-collision.js

```diff
@@ -0,0 +1,60 @@
+import Subduction from './subduction'
+import Orogeny from './orogeny'
+import VolcanicActivity from './volcanic-activity'
+
+function applyDragForces (bottomField, topField) {
+  bottomField.draggingPlate = topField.plate
+  topField.draggingPlate = bottomField.plate
+}
+
+function subduction (bottomField, topField) {
+  if (!bottomField.subduction) {
+    bottomField.subduction = new Subduction(bottomField)
+  }
+  if (!topField.volcanicAct) {
+    topField.volcanicAct = new VolcanicActivity(topField)
+  }
+  bottomField.subduction.setCollision(topField)
+  topField.volcanicAct.setCollision(bottomField)
+}
+
+function islandCollision (bottomField, topField) {
+  // Island collision (it will be merged with colliding plate).
+  topField.plate.mergeIsland(bottomField, topField)
+}
+
+function orogeny (bottomField, topField) {
+  applyDragForces(bottomField, topField)
+  if (!bottomField.orogeny) {
+    bottomField.orogeny = new Orogeny(bottomField)
+  }
+  if (!topField.orogeny) {
+    topField.orogeny = new Orogeny(topField)
+  }
+  bottomField.orogeny.setCollision(topField)
+  topField.orogeny.setCollision(bottomField)
+}
+
+export default function fieldsCollision (bottomField, topField) {
+  if (bottomField.boundary && topField.boundary) {
+    // Skip collision between field at boundary, so simulation looks a bit cleaner.
+    return
+  }
+
+  if (bottomField.isOcean) {
+    subduction(bottomField, topField)
+    if (bottomField.isContinentBuffer && topField.isOcean) {
+      // Special case when the continent is "trying" to subduct under the ocean. Apply drag force to stop both plates.
+      applyDragForces(bottomField, topField)
+    }
+  } else if (bottomField.isContinent) {
+    if (topField.isContinent) {
+      orogeny(bottomField, topField)
```

Comment:
So to clarify, this is the orogeny change - continents trying to subduct under islands now creates drag forces, instead of orogenies.

Reply:
You're right. It wasn't planned change, but this case is only theoretical at the moment. We never have islands at the boundary, as they're created in some distance from it. We don't have such presets either. And the drag forces are applied in orogeny too. The only difference is whether some small mountains will be formed or not. I don't really have an opinion on that. We can change it to `if (topField.continentalCrust) ...` if it's necessary later or it feels better to you.